main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -597,6 +599,65 @@ object EntryPoints {
 
   }
 
+  private object WrapTest {
+
+
+    /**
+      * WrapTest takes all tests and wraps them in a default handler for Assert
+      *
+      * Then, it returns a new root with these new tests
+      *
+      */
+    def run(root: TypedAst.Root)(implicit flix: Flix): TypedAst.Root = {
+      val updatedDefs = ParOps.parMapValues(root.defs)( defn =>
+        if( isTest(defn))
+          mkTest(defn, root)
+        else defn
+      )
+      root.copy(defs = updatedDefs)
+    }
+    /**
+      * mkTest takes a test function and creates a new test function with a
+      * default Assert handler by using Assert.runWithIO.
+      *
+      * Takes def oldTest(arg1:t1, ...) : ret \ ef = exp
+      * Returns def oldTest(arg1:t1, ...) : ret \ IO + Sys + (ef - Assert) = Assert.runWithIO(_ -> exp)
+      */
+    private def mkTest(oldTest: TypedAst.Def, root: TypedAst.Root)(implicit flix: Flix): TypedAst.Def = {
+      // The new type is the same as the wrapped test with an effect set of
+      // `IO + Sys + (eff - Assert)` where `eff` is the effect set of the old test.
+      val eff = Type.mkUnion(Type.IO, Type.Sys, Type.mkDifference(oldTest.spec.eff, Type.Assert, SourceLocation.Unknown)
```

Comment:
Using SourceLocation.Unknown should be avoided as much as possible. Better to reuse a source location, e.g. of the current effect.

Reply:
This comment applies everywhere you use Unknown.